### PR TITLE
net/devif/ip: build l2 header on the IP layer

### PR DESCRIPTION
**Pull request by anchao**
## Summary

net/devif/ip: build l2 header on the IP layer

## Impact

N/A

## Testing

ci check

### CODE REVIEW

File: arch/arm/src/gd32f4/gd32f4xx_enet.c [hunk 1]
```diff
@@ -1718,10 +1718,6 @@ static void gd32_receive(struct gd32_enet_mac_s *priv)
 
           if (priv->dev.d_len > 0)
             {
-              /* Update the Ethernet header with the correct MAC address */
-
-              arp_out(&priv->dev);
```
Comment (xiaoxiang781216) [inline arch/arm/src/gd32f4/gd32f4xx_enet.c:1723]: let's move arp_out/neighbor_out prototype to  net/arp/arp.h and net/neighbor/neighbor.h

File: arch/arm/src/c5471/c5471_ethernet.c [hunk 2]
```diff
@@ -1338,21 +1338,6 @@ static void c5471_receive(struct c5471_driver_s *priv)
           if (dev->d_len > 0 &&
              (EIM_TXDESC_OWN_HOST & getreg32(priv->c_rxcpudesc)) == 0)
             {
-              /* Update the Ethernet header with the correct MAC address */
```
Comment (xiaoxiang781216) [inline arch/arm/src/c5471/c5471_ethernet.c:1341]: let's move arp_ipin  at line 1320 into to ipv4_input

File: arch/arm/src/gd32f4/gd32f4xx_enet.c (hunk 1 above)
Comment (anchao) [reply arch/arm/src/gd32f4/gd32f4xx_enet.c:1723]: Depends on https://github.com/apache/nuttx/pull/7701

File: arch/arm/src/c5471/c5471_ethernet.c (hunk 2 above)
Comment (anchao) [reply arch/arm/src/c5471/c5471_ethernet.c:1341]: Done

**Review by xiaoxiang781216 — APPROVED** [review]
LGTM.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**